cfpb/consumerfinance.gov
# Filing instruction guide table of contents improvements

Latest iteration of the FIG TOC plus some misc. style improvements for FIG and browser tests to discourage regressions.

## Changes

- Rewrote the TOC to use the browser's native [Intersection Observer API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) to control the menu items.
- Moved the TOC into an expandable that sticks to the top of the viewport for mobile users.
- Bumped Django's `DATA_UPLOAD_MAX_NUMBER_FIELDS` from 2k to 3k because our FIG page is real big.

## How to test this PR

1. Units tests should pass: `yarn jest fig-sidenav-utils-spec.js`
1. Functional tests should pass headlessly:
    `yarn cypress run --spec test/cypress/integration/compliance/small-business-lending/filing-instruction-guide.js --config baseUrl=https://<demo-server-4>.cfpb.gov`
1. Functional tests should pass with your local copy of Chrome:
    `yarn cypress open --config baseUrl=https://<demo-server-4>.cfpb.gov`

## Screenshots

View it live at `https://<demo-server-4>.cfpb.gov/compliance/compliance-resources/small-business-lending/1071-filing-instruction-guide/`

## References

- https://github.local/CFPB/1071-data-collection/issues/164
- https://github.local/CFPB/1071-data-collection/issues/157
- https://github.local/CFPB/1071-data-collection/issues/163
- https://github.local/CFPB/1071-data-collection/issues/160

## Checklist



  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance

### Front-end testing



#### Browser testing

Visually tested in the following supported browsers:



#### Accessibility


#### Other
